### ISSUE TITLE
[ENH] add random_state to BaseDeepNetworkPyTorch for reproducible training

### ISSUE THREAD

**Issue by paramsureliya** [alan-turing-institute/sktime#10056]
## Problem

All deep learning forecasters inheriting from `BaseDeepNetworkPyTorch` produce non-reproducible
results because the DataLoader shuffle is never seeded. Calling `.fit()` twice with identical
data gives different predictions every time:

```python
from sktime.forecasting.scinet import SCINetForecaster
from sktime.datasets import load_airline

y = load_airline()
model = SCINetForecaster(seq_len=8)
model.fit(y, fh=[1, 2, 3])
pred1 = model.predict()

model2 = SCINetForecaster(seq_len=8)
model2.fit(y, fh=[1, 2, 3])
pred2 = model2.predict()

print(pred1.equals(pred2))  # False — different every run
```

## Affected Forecasters

- `ESRNNForecaster`
- `SCINetForecaster`
- `LTSFLinearForecaster`, `LTSFDLinearForecaster`, `LTSFNLinearForecaster`, `LTSFTransformerForecaster`
- `RBFForecaster`
- `CINNForecaster`

## Root Cause

`build_pytorch_train_dataloader` and `_build_panel_dataloader` in `BaseDeepNetworkPyTorch`
call `DataLoader` with `shuffle=True` but pass no seeded generator:

```python
return DataLoader(dataset, self.batch_size, shuffle=True)  # no seed = different every run
```

Additionally, several child classes override these dataloader methods with their own
implementations that also lack seeding, meaning a base-class-only fix is not sufficient.

## Evidence It Can Work

`ConvTimeNetForecaster` already solved this locally by passing a seeded `torch.Generator`
to its DataLoader. The fix lifts that same pattern into the base class and applies it
consistently across all overriding child classes.

## Proposed Fix

**Base class (`BaseDeepNetworkPyTorch`):**

- Add `random_state=None` parameter to `__init__`
- Seed a `torch.Generator` in `build_pytorch_train_dataloader` when `random_state is not None`
- Seed a `torch.Generator` in `_build_panel_dataloader` when `random_state is not None`
- Add `"capability:random_state": True` to `_tags`
- Use `if self.random_state is not None` (not `if self.random_state`) so that `random_state=0` is treated as a valid seed

**Child classes** — propagate `random_state` to all 8 forecasters' `__init__` signatures so
users can pass it in. Additionally, fix all overridden dataloader methods that bypass the base class:

- `ESRNNForecaster` — overrides `build_pytorch_train_dataloader` and `_build_panel_dataloader`
- `LTSFTransformerForecaster` — overrides `build_pytorch_train_dataloader` and `_build_panel_dataloader`
- `LTSFLinearForecaster`, `LTSFDLinearForecaster`, `LTSFNLinearForecaster` — override `_build_panel_dataloader`
- `RBFForecaster` — overrides both dataloader methods; also has an unseeded DataLoader inside `_fit`
- `CINNForecaster` — has unseeded DataLoaders inside `_fit`, `_pretrain`, and `_pretrain_update`

## After the Fix

```python
model = SCINetForecaster(seq_len=8, random_state=42)
model.fit(y, fh=[1, 2, 3])
pred1 = model.predict()

model2 = SCINetForecaster(seq_len=8, random_state=42)
model2.fit(y, fh=[1, 2, 3])
pred2 = model2.predict()

print(pred1.equals(pred2))  # True — reproducible every run
```
